cmd/ilm-restore.go

```diff
@@ -0,0 +1,323 @@
+		ShowDir:           DirNone,
+	}) {
+		if content.Err != nil {
+			errorIf(content.Err.Trace(client.GetURL().String()), "Unable to list folder.")
+			continue
+		}
+		err := restoreObject(ctx, targetAlias, content.URL.String(), content.VersionID, days)
+		if err != nil {
+			restoreSentReq <- err
+			continue
+		}
+		// Avoid sending the status of each separate version
+		// of the same object name.
+		if prev != content.URL.String() {
+			prev = content.URL.String()
+			restoreSentReq <- nil
+		}
+	}
+}
+
+// Wait until an object which receives restore request is completely restored in the fast tier
+func waitRestoreObject(ctx context.Context, targetAlias, targetURL, versionID string) *probe.Error {
+	clnt, err := newClientFromAlias(targetAlias, targetURL)
+	if err != nil {
+		return err
+	}
+
+	for {
+		opts := StatOptions{versionID: versionID}
+		st, err := clnt.Stat(ctx, opts)
+		if err != nil {
+			return err
+		}
+		if st.Restore == nil {
+			return probe.NewError(fmt.Errorf("`%s` did not receive restore request", targetURL))
+		}
+		if st.Restore != nil && !st.Restore.OngoingRestore {
+			return nil
+		}
+	}
+}
+
+// Check and wait the restore status of one or more objects one by one.
+func checkRestoreStatus(ctx context.Context, targetAlias, targetURL, targetVersionID string, recursive, applyOnVersions bool, restoreStatus chan *probe.Error) {
+	defer close(restoreStatus)
+
+	client, err := newClientFromAlias(targetAlias, targetURL)
+	if err != nil {
+		restoreStatus <- err
+		return
+	}
+
+	if !recursive {
+		restoreStatus <- waitRestoreObject(ctx, targetAlias, targetURL, targetVersionID)
+		return
+	}
+
+	prev := ""
+	for content := range client.List(ctx, ListOptions{
+		Recursive:         true,
+		WithOlderVersions: applyOnVersions,
+		ShowDir:           DirNone,
+	}) {
+		if content.Err != nil {
+			restoreStatus <- content.Err
+			continue
+		}
+
+		err := waitRestoreObject(ctx, targetAlias, content.URL.String(), content.VersionID)
+		if err != nil {
+			restoreStatus <- err
+			continue
+		}
+
+		if prev != content.URL.String() {
+			prev = content.URL.String()
+			restoreStatus <- nil
+		}
+	}
+}
+
+var dotCycle = 0
+
+// Clear and print text in the same line
+func printStatus(msg string, args ...interface{}) {
+	if globalJSON {
+		return
+	}
+
+	dotCycle++
+	dots := bytes.Repeat([]byte{'.'}, dotCycle%3+1)
+	fmt.Print("\n\033[1A\033[K")
+	fmt.Printf(msg+string(dots), args...)
+}
+
+// Receive restore request & restore finished status and print in the console
+func showRestoreStatus(restoreReqStatus, restoreFinishedStatus chan *probe.Error, doneCh chan struct{}) {
+	var sent, finished int
+	var done bool
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for !done {
+		select {
+		case err, ok := <-restoreReqStatus:
+			if !ok {
+				done = true
+				break
+			}
+			errorIf(err.Trace(), "Unable to send restore request.")
+			if err == nil {
+				sent++
+			}
+		case <-ticker.C:
```

Comment:
why do we need the ticker here?

Reply:
> why do we need the ticker here?

Just to print status each one second.. the status shows progress with dots so the user feels something is in progress